### PR TITLE
Cod 57-Set dynamic routes for categories page

### PR DESCRIPTION
**Pull request by NeginDn**

### CODE REVIEW

File: pages/categories/[slug].js [hunk 1]
```diff
@@ -1,25 +1,66 @@
...
+              <SubCatCard
+                key={id}
+                backgroundColor1={
+                  category_id === 1
+                    ? "yellow"
+                    : category_id === 2
+                    ? "green"
+                    : category_id === 3
+                    ? "indigo"
+                    : "indigo"
```
Comment (artcodespace) [inline pages/categories/[slug].js]: This seems a little clunky. Why not try something like `backgroundColor1={getColorForCategoryId(category_id)}` and then you could write a short function that returns the colour string depending on what argument it gets passed?

I think that would make it easier to read.

File: lib/titleConvertor.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const titleConvertor = (name) => {
```
Comment (artcodespace) [inline lib/titleConvertor.js]: Could this name be made clearer to do you think? What is this function doing? Is it converting something to or from a title?

File: pages/categories/[slug].js [hunk 3]
```diff
@@ -1,25 +1,66 @@
...
+        backgroundColor="white"
+        imgSrc={headerImgCat}
+        imgAlt="Woman's hand"
+        title={titleConvertor(slug)}
+        caption={
+          "Miricyl helps you find key, informative resources to guide you on your mental health journey"
+        }
+      />
+      <div className="flex flex-col md:flex-row md:flex-wrap justify-end items-center md:justify-between my-12 sm:my-32 sm:mx-12">
+        {issuesCategories &&
```
Comment (artcodespace) [inline pages/categories/[slug].js]: I don't think you need to check `issuesCategories` here, you already know that if `!issuesCategories` you'll hit the return on line 19

**Comment by NeginDn** [comment]
Hi @ARTurnerGit,
 Very interesting feedbacks. They all make the code way cleaner. Thank you. I applied them all

File: components/CategorySelection/index.js [hunk 4]
```diff
@@ -27,9 +28,9 @@ const CategoriesSelection = () => {
             <Card
               key={id}
               id={id}
-              title={name}
+              title={convertSlugToTitle(name)}
```
Comment (artcodespace) [inline components/CategorySelection/index.js]: The name of this function is way better. You can probably guess my next question now. If the function converts a slug to a title, why are you passing it the name and not the slug here?

File: lib/getColorForCategoryId.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  if (categoryId === 1) {
+    return "yellow";
+  } else if (categoryId === 2) {
+    return "green";
+  } else if (categoryId === 3) {
+    return "indigo";
+  } else {
+    return "red";
+  }
+};
```
Comment (artcodespace) [inline lib/getColorForCategoryId.js]: Absolutely fine, may be clearer as a switch statement but it wouldn't change the behaviour

File: components/CategorySelection/index.js (hunk 4 above)
Comment (artcodespace) [reply components/CategorySelection/index.js]: It looks to me that all the other calls to this function are called with a slug

**Comment by artcodespace** [comment]
Don't forget to resolve the conflicts as well Negin

File: components/CategorySelection/index.js (hunk 4 above)
Comment (NeginDn) [reply components/CategorySelection/index.js]: Makes perfect sense,  Done :)

**Review by iTrip23 — APPROVED** [review]

**Pull request merged**